tests/test_utils_url.py

```diff
@@ -73,6 +73,22 @@ class MySpider(Spider):
         self.assertTrue(url_is_from_spider('http://www.example.net/some/page.html', MySpider))
         self.assertFalse(url_is_from_spider('http://www.example.us/some/page.html', MySpider))
 
+    def test_add_scheme_if_missing(self):
+        self.assertEqual(add_scheme_if_missing('http://www.example.com'),
+                                               'http://www.example.com')
+        self.assertEqual(add_scheme_if_missing('http://www.example.com/some/page.html'),
+                                               'http://www.example.com/some/page.html')
+        self.assertEqual(add_scheme_if_missing('http://example.com'),
+                                               'http://example.com')
+        self.assertEqual(add_scheme_if_missing('www.example.com'),
+                                               'http://www.example.com')
+        self.assertEqual(add_scheme_if_missing('example.com'),
+                                               'http://example.com')
+        self.assertEqual(add_scheme_if_missing('//example.com'),
+                                               'http://example.com')
+        self.assertEqual(add_scheme_if_missing('https://www.example.com'),
+                                               'https://www.example.com')
```

Comment:
Could you please add tests for URLs with 

* some variations of port, fragment, GET arguments;
* URLs without scheme but with `//` separator, like `//example.com/foo/bar`.